tests/components/light/test_init.py

```diff
@@ -2,495 +2,623 @@
-        assert not light.is_on(self.hass, ent2.entity_id)
-        assert not light.is_on(self.hass, ent3.entity_id)
+    assert not light.is_on(hass, ent1.entity_id)
+    assert not light.is_on(hass, ent2.entity_id)
+    assert not light.is_on(hass, ent3.entity_id)
 
-        # toggle all lights
-        common.toggle(self.hass)
+    # toggle all lights
+    await hass.services.async_call(
+        light.DOMAIN, SERVICE_TOGGLE, {ATTR_ENTITY_ID: ENTITY_MATCH_ALL}, blocking=True
+    )
 
-        self.hass.block_till_done()
+    assert light.is_on(hass, ent1.entity_id)
+    assert light.is_on(hass, ent2.entity_id)
+    assert light.is_on(hass, ent3.entity_id)
 
-        assert light.is_on(self.hass, ent1.entity_id)
-        assert light.is_on(self.hass, ent2.entity_id)
-        assert light.is_on(self.hass, ent3.entity_id)
+    # toggle all lights
+    await hass.services.async_call(
+        light.DOMAIN, SERVICE_TOGGLE, {ATTR_ENTITY_ID: ENTITY_MATCH_ALL}, blocking=True
+    )
 
-        # toggle all lights
-        common.toggle(self.hass)
+    assert not light.is_on(hass, ent1.entity_id)
+    assert not light.is_on(hass, ent2.entity_id)
+    assert not light.is_on(hass, ent3.entity_id)
 
-        self.hass.block_till_done()
+    # Ensure all attributes process correctly
+    await hass.services.async_call(
+        light.DOMAIN,
+        SERVICE_TURN_ON,
+        {
+            ATTR_ENTITY_ID: ent1.entity_id,
+            light.ATTR_TRANSITION: 10,
+            light.ATTR_BRIGHTNESS: 20,
+            light.ATTR_COLOR_NAME: "blue",
+        },
+        blocking=True,
+    )
+    await hass.services.async_call(
+        light.DOMAIN,
+        SERVICE_TURN_ON,
+        {
+            ATTR_ENTITY_ID: ent2.entity_id,
+            light.ATTR_RGB_COLOR: (255, 255, 255),
+            light.ATTR_WHITE_VALUE: 255,
+        },
+        blocking=True,
+    )
+    await hass.services.async_call(
+        light.DOMAIN,
+        SERVICE_TURN_ON,
+        {
+            ATTR_ENTITY_ID: ent3.entity_id,
+            light.ATTR_XY_COLOR: (0.4, 0.6),
+        },
+        blocking=True,
+    )
 
-        assert not light.is_on(self.hass, ent1.entity_id)
-        assert not light.is_on(self.hass, ent2.entity_id)
-        assert not light.is_on(self.hass, ent3.entity_id)
+    _, data = ent1.last_call("turn_on")
+    assert data == {
+        light.ATTR_TRANSITION: 10,
+        light.ATTR_BRIGHTNESS: 20,
+        light.ATTR_HS_COLOR: (240, 100),
+    }
 
-        # Ensure all attributes process correctly
-        common.turn_on(
-            self.hass, ent1.entity_id, transition=10, brightness=20, color_name="blue"
-        )
-        common.turn_on(
-            self.hass, ent2.entity_id, rgb_color=(255, 255, 255), white_value=255
-        )
-        common.turn_on(self.hass, ent3.entity_id, xy_color=(0.4, 0.6))
+    _, data = ent2.last_call("turn_on")
+    assert data == {light.ATTR_HS_COLOR: (0, 0), light.ATTR_WHITE_VALUE: 255}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)